### DISCUSSION TITLE
dropdown form changes according to the value

### DISCUSSION THREAD

**Discussion by erakastrati** [Budibase/budibase#2483] [Help] [answered]
Hi there,
I have such a task: 
- The form named Gjykata has some dropdown for different states.
<img width="1282" alt="Screen Shot 2021-08-27 at 1 37 33 PM" src="https://user-images.githubusercontent.com/74623805/131121832-ab375067-be6b-4e0d-83c3-228ac6429a51.png">

 You have to choose one, for example if you choose America the places of America are shown in the second form.
<img width="910" alt="Screen Shot 2021-08-27 at 1 41 23 PM" src="https://user-images.githubusercontent.com/74623805/131122102-c47d7b03-67b6-4f94-bb0c-12cfb99b9a81.png">
Thus we can say. that the second form is dependent from the first one. 
Is there any idea how to make it possible?
Thanks, 
Era.

**Comment by EronitaB** [comment]
@joebudi @kevmodrome @shogunpurple I have the same problem! Any idea?

**Comment by mjashanks** [comment, marked as answer]
We are currently working on this. 

It will be available in the next release - over the next few business days!

**Reply by erakastrati** [reply]
All right, thanks @mjashanks !

**Comment by ChechaTocon** [comment]
I'm so interested in this feature, is it already released?, if so where can I find it?

Thanks in advanced

**Reply by ronaldbalza** [reply]
Count me in. Is this some sort of a cascading dropdown? :)

**Reply by mjashanks** [reply]
This is released already. I need to write up documentation for it.

In my video below, I have 
1. A "Body Site" dropdown, with custom options "Arm" and "Leg"
2. A dataprovider that reads from my "Diagnosis" table, filtered by the selected Body Site
3. A Diagnosis dropdown, which loads from the data provider

Note that the data provider can load from anything - a table, a SQL Db, REST API etc.

https://user-images.githubusercontent.com/3524181/139043505-5a5bd07e-1bc7-43d6-b409-daa790db18e2.mp4

**Reply by ronaldbalza** [reply]
@mjashanks Cool cool cool! Thanks for the heads-up shanks!

**Reply by MrWombastic** [reply]
Hey. I got a question to this feature. I am having the same schema as you regarding the 2 tables, but the difference is, that the column I want to filter the second dropdown isnt type text, it's relationship. For example:
Table SubAssets:
![image](https://user-images.githubusercontent.com/34077172/167927747-51af6d36-47de-4d61-84b0-3eb906fcc8a9.png)
Table Assets:
![image](https://user-images.githubusercontent.com/34077172/167927934-361a31ba-8b44-4808-8499-1087e0d1a8d6.png)

So first option picker:
I pick "Silber"
Second option picker should only show values associated with the Asset "Silber" (1st screenshot).

Filtering the data provider for the SubAsset option picker (2nd) seems to be a problem, because column with type relationship is not visible:
![image](https://user-images.githubusercontent.com/34077172/167928285-914d30f8-890b-4f9d-b5cd-9d832edb0289.png)

Am I wrong?

**Reply by MrWombastic** [reply]
I now saw that your first dropdown isn't based on the diagnostics table, instead of choosing the values from this table, you make an option field with fixed values to choose of. That's ok for a short formula, but if you got 50 options in our table, you wont update the dropdown manually every time.
Think of a table where a user could enter new data to fill the dropdown, I don't see how to realize it with this solution.
Am I wrong?

**Comment by NetLancer** [comment]
Hi guys! I've been wondering if we could take it a step (or few steps) further: I came from MS Access world where all the entities are in their own tables. So here's a table (for which i want to build an interface) "Sales" which consists of a number of fields and some of the fields are dropdowns of separate tables (entities). For example: "Offices" & "Employees", which are interconnected by "One-to-many", and  "Consolidators" & "Carriers" interconnected by "Many-to-many", meaning - One Consolidator may have many carriers and vice versa. Now all of those 4 tables(entities) are selectable in the main table "Sales", which looks like following fields sequence: [ Date, Office, Employee, Consolidator, Carrier, etc etc.. ]. 
I'm struggling to find the way to make fields "Employees" & "Carriers" dependent on "Offices" & "Consolidators" selected options accordingly, and it should be done in an Interface which saves the data to "Sales". 
I believe it should be possible using BB's built in database but so far can't figure how. 
Need help(!), i'll post more detailed picture of tables with field if needed but i think the problem's clear, and generally i think this sort of challenges (like interdependent dropdown lists) should be addressed somehow in documentation if the Tool is capable of implementing it. Appreciate it

**Reply by MrWombastic** [reply]
Hey,
I've watched the video of @mjashanks and I think the problem @NetLancer describes is same as mine, and for someone who comes from relational database models, it is very confusing.

I got two tables, Assets and SubAssets.
When you enter a new row, you MUST select an Asset, and you CAN select a SubAsset which is dependent on the chosen Asset.

Therefore I got two tables:
Assets: Id, AssetName
SubAssets: Id, SubAssetName, AssetID where the AssetID is equal to the Id from the table Assets.

Assets Table
![grafik](https://user-images.githubusercontent.com/34077172/184506323-675923df-378d-41d3-8a4b-3763ab6be3f9.png)
Subassets Table
![grafik](https://user-images.githubusercontent.com/34077172/184506343-4133ddc1-7ec9-4a9b-8769-1eb493615d4d.png)

Now when I build what you built in your video @mjashanks, it doesn't work with relationship-columns (Screenshot2: Asset-Column in the SubAssets Table), because such a column isn't choosable in the DataProvider. It also doesn't work with a separate column in the SubAssets Table, where the AssetName is entered as text (Screenshot2: Assetname-Column in the SubAssets Table).

![grafik](https://user-images.githubusercontent.com/34077172/184506448-879191a1-e477-43c6-82d5-783fa4a5c217.png)

![grafik](https://user-images.githubusercontent.com/34077172/184506509-616d2885-4e06-4cd5-89ed-a5e95576c880.png)
![grafik](https://user-images.githubusercontent.com/34077172/184506493-cd7f03b8-461c-4b18-823b-284ed904274e.png)

With a dropdown where the values arent based on a table but entered in the formula itself like in your video, it works. But only the functionality of the filter for the 2nd field works. In this case I get the problem that in my case AssetID and SubAssetID shall be saved to the table where the Transaction ist entered. In this table the fields are relationship fields, of course. I cannot safe the manually filled assetID Dropdown to a relationship field.
And even if, what if someone enters a completly new asset in the database, the dropdown wouldn't be updated because it doesn't rely on a table.

BUT: wherefore do I need relations between tables anymore, if they aren't useful for such things? Right now it feels like having relationships is useless.

Am I missing something obvious or why is it designed like that?

Best regards

Markus

**Reply by NetLancer** [reply]
Yeah, i think BB's documentation is lacking clear examples of Relational DB implementations where budibase serves an interface, and i believe relational model implementations still have more use cases and are in demand
